shell_completions.md

```diff
@@ -431,4 +439,38 @@ search  show  status
 
 ## PowerShell completions
 
-Please refer to [PowerShell Completions](powershell_completions.md) for details.
+Cobra supports native PowerShell completions generated from the root `cobra.Command`. You can use the `command.GenPowerShellCompletion()` or `command.GenPowerShellCompletionFile()` functions. You must provide these functions with a parameter indicating if the completions should be annotated with a description. Cobra will provide the description automatically based on usage information. You can choose to make this option configurable by your users.
+
+The script is designed to support all three Powershell completion modes:
+* TabCompleteNext (default windows style - on each key press the next option is displayed)
+* Complete (works like bash)
+* MenuComplete (works like zsh)
+You set the mode with `Set-PSReadLineKeyHandler -Key Tab -Function <mode>`. Descriptions are only displayed when using the `Complete` or `MenuComplete` mode.
```

Comment:
Need a newline before this line.  GitHub currently puts on the same line as the last bullet, can be seen here:
https://github.com/Luap99/cobra/blob/powershell-custom-comp/shell_completions.md